### ISSUE TITLE
'DirectoryIterator' object has no attribute 'nb_sample'

### ISSUE THREAD

**Issue by yanyongwei** [keras-team/keras#14167]
**Important note:** the development of Keras is currently taking place at [`github.com/tensorflow/tensorflow`](https://github.com/tensorflow/tensorflow). Please file your bug report by creating a new issue in the TensorFlow repository.

`
def write_gap(MODEL, image_size, lambda_func=None):
    width = image_size[0]  
    height = image_size[1]  
    input_tensor = Input((height, width, 3))
    x = input_tensor
    if lambda_func:
        x = Lambda(lambda_func)(x)
    base_model = MODEL(input_tensor=x, weights='imagenet', include_top=False)
    model = Model(base_model.input, GlobalAveragePooling2D()(base_model.output))

    gen = ImageDataGenerator()
    train_generator = gen.flow_from_directory("data/train2", image_size, shuffle=False,
                                              batch_size=16)
    test_generator = gen.flow_from_directory("data/test2", image_size, shuffle=False,
                                             batch_size=16, class_mode=None)
    train = model.predict_generator(train_generator, train_generator.nb_sample)
    test = model.predict_generator(test_generator, test_generator.nb_sample)
    with h5py.File("gap_%s.h5"%MODEL.func_name) as h:
        h.create_dataset("train", data=train)
        h.create_dataset("test", data=test)
        h.create_dataset("label", data=train_generator.classes)

write_gap(ResNet50, (224, 224))
`

**Comment by jvishnuvardhan** [comment]
@yanyongwei Can you please share a simple standalone code to reproduce the issue? Thanks!
